packages/server/api/src/app/flows/issues/issues-service.ts

```diff
@@ -57,20 +77,71 @@ export const issuesService = (log: FastifyBaseLogger) => ({
             },
         })
 
-        const query = repo().createQueryBuilder(IssueEntity.options.name).where({
-            projectId,
-            status: IssueStatus.ONGOING,
-        })
+        let query = repo().createQueryBuilder(IssueEntity.options.name)
+            .where({ projectId })
+
+        if (status && status.length > 0) {
+            query = query.andWhere('status IN (:...statuses)', { statuses: status })
+        }
 
         const { data, cursor: newCursor } = await paginator.paginate(query)
 
-        const populatedIssues = await Promise.all(data.map(async issue => {
+        const issuesWithCounts = await Promise.all(data.map(async issue => {
+            // Fixed query with proper join between issue and flow_run
+            const count = await repo()
```

Comment:
I think this function joins flow_run with issues but that really no need for the joining at all.

It's simple count by flowId, failedStepName and failed status.

```
    const count = await flowRunRepo().countBy({
        flowId: issue.flowId,
        failedStepName: issue.stepName,
        status: In([FlowRunStatus.FAILED, FlowRunStatus.INTERNAL_ERROR, FlowRunStatus.TIMEOUT]),
    })
    ```